mgr/dashboard_v2: add  sparkline and monitor  components

tspmelo opened the pull request:
This PR includes both the sparkline and the monitor components.

You can see both components in the image below, 
the sparkline is being reused in each row of the table:
![192 168 1 136_4200_ fhd 1](https://user-images.githubusercontent.com/399326/35700664-7faf656c-078b-11e8-8490-024eb9e738ef.png)

New version:
![192 168 1 136_4200_ fhd 2](https://user-images.githubusercontent.com/399326/35883399-0cc8a1e2-0b7f-11e8-973d-56f331baa534.png)

`Signed-off-by: Tiago Melo <tmelo@suse.com>`

ricardoasmarques commented on src/pybind/mgr/dashboard_v2/tests/test_monitor.py:
```diff
@@ -0,0 +1,18 @@
+    def test_ping(self):
+        self.getPage("/api/monitor")
+        self.assertStatus('200 OK')
+        # self.assertJsonBody({'echo': 'Hello World'})
```
Remove this line

sebastian-philipp replied:
What about actually testing the body? (and rename test_ping)

jecluis commented:
So, if this PR requires the thing you introduced in #57, and #57 is not actually making use of said code, why not submitting those commits alongside this PR?

tspmelo commented:
merged #57 into this PR.

Devp00l commented on src/pybind/mgr/dashboard_v2/frontend/src/app/ceph/cluster/monitor/monitor.component.html:
```diff
@@ -0,0 +1,115 @@
+</h1>
+
+<!-- Main content -->
+<div class="row">
```
Should be replace with the data table component.

tspmelo replied:
I'm now using the cdTable on both quorum tables.

Devp00l commented on src/pybind/mgr/dashboard_v2/frontend/src/app/shared/components/sparkline/sparkline.component.ts:
```diff
@@ -0,0 +1,76 @@
+  ngOnInit() {}
+
+  ngOnChanges(changes: SimpleChanges) {
+    this.datasets[0].data = changes['data'].currentValue;
```
Is "currentValue" an array?

tspmelo replied:
In this case, yes.
It represents the current value of the input `data`

Devp00l commented on src/pybind/mgr/dashboard_v2/frontend/src/app/shared/components/sparkline/sparkline.component.ts:
```diff
@@ -0,0 +1,76 @@
+
+  ngOnChanges(changes: SimpleChanges) {
+    this.datasets[0].data = changes['data'].currentValue;
+    this.datasets = [...this.datasets];
```
Is this needed to get an update from Angular?

tspmelo replied:
It seems like it, if you don't use the `...` the plugins will not update.

jecluis replied:
@tspmelo mind elaborating a bit? :)

ricardoasmarques commented:
Shouldn't we add a breadcrumb like we have on other pages (e.g. Hosts / Pools)?

ricardoasmarques commented on src/pybind/mgr/dashboard_v2/frontend/src/app/ceph/cluster/monitor-sparkline-data.pipe.ts:
```diff
@@ -0,0 +1,16 @@
+})
+export class MonitorSparklineDataPipe implements PipeTransform {
+  transform(value: any, args?: any): any {
+    // let result = '';
```
Commented code should be removed.

Commit:
mgr/dashboard_v2: add performance counter page

Signed-off-by: Tiago Melo <tmelo@suse.com>

Commit:
mgr/dashboard_v2: move datatable module

By moving the datatable module outside of the  components module we
can avoid a circular dependency between the datatable and sparkline
component.

Signed-off-by: Tiago Melo <tmelo@suse.com>

Commit:
mgr/dashboard_v2: add sparkline component

This also adds a template in the table component.

Signed-off-by: Tiago Melo <tmelo@suse.com>

Commit:
mgr/dashboard_v2: add monitors page

Signed-off-by: Tiago Melo <tmelo@suse.com>

Commit:
mgr/dashboard_v2: add router link table template

Signed-off-by: Tiago Melo <tmelo@suse.com>

tspmelo commented:
Just pushed a new version.

Changes:
- Applied suggested changes
- Add 2 new table templates, sparkline and router link.
- Add new performance counter page, so we can have a dedicated page to show that information
- Move datatable module to another module, to fix a circular dependency error

votdev commented:
When the browser window is small, the paging input field in the datatable tool header shouldn't be resized. There should be displayed the number '10'.
![bildschirmfoto vom 2018-02-15 14-15-55](https://user-images.githubusercontent.com/1897962/36258433-ebd71882-125a-11e8-889a-acf019fedbf4.png)

 ```
--- src/pybind/mgr/dashboard_v2/frontend/src/app/shared/datatable/table/table.component.scss	(date 1518694121000)
+++ src/pybind/mgr/dashboard_v2/frontend/src/app/shared/datatable/table/table.component.scss	(date 1518700914000)
@@ -76,6 +76,7 @@
   }
   .input-group.dataTables_paginate {
     width: 8%;
+    min-width: 75px;
   }
 }
 ```
